src/Augmint.ts

```diff
@@ -205,13 +200,128 @@ export class Augmint {
                 throw new Error("legacy contracts length mismatch!");
             }
         }
-        const options: ILoanManagerOptions = {
-            token: this.token, // FIXME: This should come from LoanManager contract's augmintToken property
-            rates: this.rates, // FIXME: This should come from LoanManager contract's rates property
-            ethereumConnection: this.ethereumConnection
-        };
         return legacyContracts.map(
-            (contract: DeployedContract<LoanManagerInstance>) => new LoanManager(contract.connect(this.web3), options)
+            (contract: DeployedContract<LoanManagerInstance>) =>
+                new LoanManager(contract.connect(this.web3), this.ethereumConnection)
         );
     }
+
+    public async getLoansForAccount(userAccount: string): Promise<Loan[]> {
+        const result: Loan[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            const lmLoans: Loan[] = await loanManager.getLoansForAccount(userAccount);
+            result.push(...lmLoans);
+        }
+        return result;
+    }
+
+    public async getLoanProducts(activeOnly: boolean): Promise<LoanProduct[]> {
+        const result: LoanProduct[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            let products: LoanProduct[];
+            if (activeOnly) {
+                products = await loanManager.getActiveProducts();
+            } else {
+                products = await loanManager.getAllProducts();
+            }
+            result.push(...products);
+        }
+        return result;
+    }
+
+    public async repayLoan(loan: Loan, repaymentAmount: Tokens, userAccount: string): Promise<Transaction> {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loan.loanManagerAddress)
+        const tokenAddress: string = await loanManager.tokenAddress;
+        const tokenContract: DeployedContract<TokenAEur> = this.deployedEnvironment.getContractFromAddress(
+            AugmintContracts.TokenAEur,
+            tokenAddress
+        );
+        const token: AugmintToken = new AugmintToken(tokenContract.connect(this.web3), this.ethereumConnection);
+        return loanManager.repayLoan(loan, repaymentAmount, userAccount, token);
+    }
+
+    public async newEthBackedLoan(loanProduct: LoanProduct, weiAmount: Wei, userAccount: string, minRate?:Tokens): Promise<Transaction> {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loanProduct.loanManagerAddress)
+        return loanManager.newEthBackedLoan(loanProduct, weiAmount, userAccount, minRate);
+    }
+
+    public async getAllLoans(): Promise<Loan[]> {
+        const result: Loan[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            const lmLoans: Loan[] = await loanManager.getAllLoans();
+            result.push(...lmLoans);
+        }
+        return result;
```

Comment:
use `Promise.all` to do requests in parallel, instead of sequentially `await`-ing.

Reply:
done